identity/aziot-identityd/src/identity.rs

```diff
@@ -522,12 +523,20 @@ impl IdentityManager {
                         identity_cert,
                         identity_pk,
                     } => {
-                        self.create_identity_cert_if_not_exist_or_expired(
-                            &identity_pk,
-                            &identity_cert,
-                            &registration_id,
-                        )
-                        .await?;
+                        let cert = self
+                            .create_identity_cert_if_not_exist_or_expired(
+                                &identity_pk,
+                                &identity_cert,
+                                IdentitySubject::RegistrationId(
+                                    registration_id.as_deref().map(Into::into),
+                                ),
+                            )
+                            .await?;
+                        let cert_cn = extract_cn(&cert)?;
+
+                        if cert_cn.is_empty() && registration_id.map_or(false, |s| !s.is_empty()) {
+                            // ???
```

Comment:
@massand I think I recall you mentioning that this should be special cased?

Reply:
Judging by DPS's behavior of rejecting empty CN certs, the first check is sufficient to error this out and fail the X.509 attestation. Initially, I thought registration_id would be used in the JSON payload when CN is missing. However, we should rename `registration_id` to `device_id_cert_subject_name` and request a cert.